src/hyperlight_guest_bin/src/lib.rs

```diff
@@ -139,11 +141,50 @@ pub static mut OS_PAGE_SIZE: u32 = 0;
 // to satisfy the clippy when cfg == test
 #[allow(dead_code)]
 fn panic(info: &core::panic::PanicInfo) -> ! {
-    let msg = info.to_string();
-    let c_string = alloc::ffi::CString::new(msg)
-        .unwrap_or_else(|_| alloc::ffi::CString::new("panic (invalid utf8)").unwrap());
+    _panic_handler(info)
+}
+
+static PANIC_BUF: Mutex<String<512>> = Mutex::new(String::new());
+
+#[inline(always)]
+fn _panic_handler(info: &core::panic::PanicInfo) -> ! {
+    let mut panic_buf_guard = PANIC_BUF.lock();
+    let write_res = write!(panic_buf_guard, "{}", info);
+    if write_res.is_err() {
+        unsafe {
+            abort_with_code_and_message(
+                &[ErrorCode::UnknownError as u8],
+                c"panic: message format failed".as_ptr(),
+            )
+        }
+    }
 
-    unsafe { abort_with_code_and_message(&[ErrorCode::UnknownError as u8], c_string.as_ptr()) }
+    let append_res = panic_buf_guard.push('\0');
```

Comment:
I think the nullbyte can be combined with the `write!` call above

Reply:
Ah good catch!